README.md

```diff
@@ -3,9 +3,22 @@
 ![Testing nbclassic](https://github.com/jupyterlab/nbclassic/workflows/Testing%20nbclassic/badge.svg)
 [![Documentation Status](https://readthedocs.org/projects/nbclassic/badge/?version=latest)](https://nbclassic.readthedocs.io/en/latest/?badge=latest)
 
-NBClassic runs the [Jupyter NbClassic](https://github.com/jupyter/nbclassic) frontend on the Jupyter Server backend.
+NBClassic is the classic Jupyter Notebook application (with the classic
+Javascript interface) running on Jupyter Server.
 
-This project prepares for a future where JupyterLab and other frontends switch to [Jupyter Server](https://github.com/jupyter/jupyter_server/) for their Python Web application backend. Using this package, users can launch Jupyter NbClassic, JupyterLab and other frontends side-by-side on top of the new Python server backend.
+This project provides compatibility with the old notebook interface and
+support during an intermediate transition period to Notebook 7, for users
+with extensions and other customizations that cannot yet be upgraded
+to the new version of Notebook.
+
+In the future, Jupyter frontends like NBClassic and JupyterLab will coexist
+side-by-side on top of a [Jupyter Server] backend. This package makes that
+possible: You can install this package beside an existing JupyterLab or
+Jupyter Notebook 7 environment, and get the best of both worlds:
```

Comment:
This paragraph feels a bit awkward to me. This project doesn't provide compatibility _with_ the old notebook interface. It _is_ the old notebook interface (stated in the first sentence above. 

I would propose that we change the language to explain exactly what the user is getting. 

> As the Notebook package evolves from v6 to v7, it redefines the notebook extension interface. As a side effect, it requires that all extensions be redeveloped to work in this new world.
> 
> Because we understand that some users' work might depend deeply on extensions that haven't yet been ported, we created this package, NBClassic. NBClassic provides the classic Notebook interface and preserves the old experience—including your classic extensions and customizations—under a new set of URL endpoints.
> 
> This package enables you to fearlessly upgrade to the new Notebook and JupyterLab frontend, while preserving your classic notebook experience under the NBClassic namespace, '/nbclassic`. All three experiences can be run side-by-side on top of the [Jupyter Server] backend.

Maybe mentioned endpoints/namespaces feels a bit technical. But I think it would help people discover nbclassic if they knew exactly what needs to change in their URL to find their old experience after an upgrade.

Reply:
Here's what I wanted to achieve with the original additions and restructuring:

- Restructure to communicate the WHY quicker (why does this project exist? what is it for?)
- Don't make people read a novel about the depths and intricacies of the Jupyter ecosystem and the long, complex history of its development, to arrive at that understanding...provide an explanation of the purpose earlier/up front

The existing Jupyter ecosystem and offerings are already very complex and difficult to understand, especially for end users who aren't Jupyter developers (and are not steeped in complex details we're exposed to everyday and take for granted), like data science or general programming users.

I appreciate the points you made in your suggestions, so I'm working on some edits that incorporate those. I do think some of the technical details are too much and too early in the text, so I think we can still include them but move them down in a way that better follows the inverted pyramid.

Reply:
In particular, we need to be careful about how we communicate certain points to people who read this:

- NbClassic is a Jupyter frontend (what does that mean?)
- Jupyter Server is the new backend (what does that mean?)
- Jupyter Notebook uses a client/server architecture (what does that mean?)
- The notebook extension interface has been redefined (most people should not need to know or care about this)
- What is an endpoint?
- What is a namespace?

This document should definitely be accessible to non-developers and other end users who want their old stuff to work because the newest upgrades broke their environments. As such, we need to be mindful about including details like these that we take for granted as software engineers, that alienate and confuse people who don't need to understand them (because the Jupyter user community is vast and diverse).

Reply:
I think the README has a slightly different purpose and audience.

Generally, the audience _is_ the "developer". The README is a plain text file found in a project's source code; its very location makes it discoverable mostly by a technical audience. When folks come to a README, I think they expect a quick synopsis of what source code is providing. 

Sure, "non-developers" might land on this page too, but (my guess) is that they would quickly look for the link to the more user-friendly documentation page to see a less technical description of the project.

I agree that your bulleted list needs some documention for the "non-developer", but those belong under the documentation pages: https://nbclassic.readthedocs.io/en/latest/

 For nbclassic, I think the README should highlight two things:
* This source code is just the new home for the classic Notebook UI source code
* It provides a solution for folks that are looking to preserve their old notebook setup.

Reply:
BTW, no need to block on me here—just making some suggestions. I'm happy with however you decide to proceed.

Reply:
@Zsailer @echarles I've refactored based on your feedback, take another look?